remote_file_system/server.py

```diff
@@ -46,28 +46,37 @@ def __init__(
         # store request id as key. value is the Message
         self.message_history: Dict[UUID, Message] = {}
         self.invocation_semantics = invocation_semantics
+        self.keep_listening = True
+
+    def stop_listening(self) -> None:
+        self.keep_listening = False
 
     def listen_for_messages(self) -> None:
         sock = socket(AF_INET, SOCK_DGRAM)
         server_address: Tuple[str, int] = (str(self.server_ip_address), self.server_port_number)
         sock.bind(server_address)
 
+        sock.settimeout(10)
```

Comment:
If we set a socket timeout like this, will the server still listen for more than 10 seconds?

Reply:
no. the server will listen for 10 seconds for any incoming messages. If there is no incoming messages, the connection is reset. But because we have the while loop, it will go into another round of waiting for 10 seconds until the stop listening function is invoked.